pkg/pdutil/pd.go

```diff
@@ -143,20 +146,34 @@ func pdRequest(
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	resp, err := cli.Do(req)
-	if err != nil {
-		return nil, errors.Trace(err)
+	var resp *http.Response
+	for i := 0; i < pdRequestRetryTime; i++ {
+		resp, err = cli.Do(req)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if resp.StatusCode != http.StatusOK {
+			if resp.StatusCode >= 500 {
+				log.Warn("pd request fail, retry", zap.Int("retry time", i), zap.String("State", resp.Status))
+				time.Sleep(time.Second)
```

Comment:
Seems after the retry, the old response won't be closed? 🤔

Reply:
fixed